mysql-test/main/mdev-32395.test

```diff
@@ -0,0 +1,13 @@
+create table t1 (c1 int) engine=myisam;
+insert into t1 values (1), (2);
+create table t2 (c2 int) engine=myisam;
+insert into t2 values (1), (2);
+create table t3 (c3 int) engine=myisam;
+insert into t3 values (1), (2);
+ 
+set optimizer_switch=
+'condition_pushdown_for_derived=off,condition_pushdown_for_subquery=off,condition_pushdown_from_having=off';
+ 
+select t1.c1 as a from t2, t1 where t1.c1=t2.c2
```

Comment:
since its a single `select` statement as a test, and to save resetting it back:
 
 ```
set statement optimizer_switch='condition_pushdown_for_derived=off,condition_pushdown_for_subquery=off,condition_pushdown_from_having=off' for
select  .....
```